### PR TITLE
chore (sdk.v2): Error out on using InputPath placeholder for artifacts require importer.

### PR DESCRIPTION
**Pull request by chensun**
**Description of your changes:**
- Importer node will be auto attached only if InputUri placeholder is used.

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?

    [Learn more about cherry-picking updates into the release branch](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#cherry-picking-pull-requests-to-release-branch).

### CODE REVIEW

File: sdk/python/kfp/v2/dsl/component_bridge.py [hunk 1]
```diff
@@ -147,6 +147,15 @@ def _input_artifact_path_placeholder(input_key: str) -> str:
       raise TypeError(
           'Input "{}" with type "{}" cannot be paired with InputPathPlaceholder.'
           .format(input_key, inputs_dict[input_key].type))
+    elif input_key in importer_spec:
+      raise TypeError(
+          'Input "{}" with type "{}" is not connected to any upstream output. '
+          'But it is used with InputPathPlaceholder. '
```
Comment (numerology) [inline sdk/python/kfp/v2/dsl/component_bridge.py:153]:
```suggestion
          'However it is used with InputPathPlaceholder. '
```

**Review by numerology — COMMENTED** [review]
Thanks @chensun !

Just one nit-picky comment, otherwise LGTM :)

File: sdk/python/kfp/v2/dsl/component_bridge.py (hunk 1 above)
Comment (chensun) [reply sdk/python/kfp/v2/dsl/component_bridge.py]: Done.

**Pull request merged**